addons/account/models/account_payment_method.py

```diff
@@ -0,0 +1,59 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models
+
+
+class AccountPaymentMethod(models.Model):
+    _name = "account.payment.method"
+    _description = "Payment Methods"
+
+    name = fields.Char(translate=True, required=True)
+    code = fields.Char(required=True)  # For internal identification
+    category = fields.Selection(selection=[('manual', 'Manual')], required=True)
+    payment_type = fields.Selection(selection=[('inbound', 'Inbound'), ('outbound', 'Outbound')], required=True)
+    journal_type = fields.Selection(
+        selection=[('bank', 'Bank'), ('cash', 'Cash')],
+        help='Choose on which journal type this method will be usable. Leave empty to allow both types.',
+        required=False
+    )
+
+    _sql_constraints = [
+        ('name_code_unique', 'unique (code, payment_type)', 'The combination code/payment type already exists!'),
+    ]
+
+
+class AccountPaymentMethodLine(models.Model):
+    _name = "account.payment.method.line"
+    _description = "Payment Methods"
+
+    # == Business fields ==
+    name = fields.Char(compute='_compute_name', readonly=False, store=True)
+    sequence = fields.Integer(default=10)
+    payment_method_id = fields.Many2one(
+        string='Payment Method',
+        comodel_name='account.payment.method',
+        domain="[('payment_type', '=?', payment_type), ('id', 'in', available_payment_method_ids)]",
+        required=True
+    )
+    payment_account_id = fields.Many2one(
+        comodel_name='account.account',
+        check_company=True,
+        copy=False,
+        ondelete='restrict',
+        domain=lambda self: "[('deprecated', '=', False), "
+                            "('company_id', '=', company_id), "
+                            "('user_type_id.type', 'not in', ('receivable', 'payable')), "
+                            "('user_type_id', '=', %s)]" % self.env.ref('account.data_account_type_current_assets').id
+    )
+    journal_id = fields.Many2one(comodel_name='account.journal', required=True, ondelete="cascade")
+
+    # == Display purpose fields ==
+    code = fields.Char(related='payment_method_id.code', readonly=True)
+    payment_type = fields.Selection(related='payment_method_id.payment_type', readonly=True)
```

Comment:
You don't need to specify ´readonly=True´ for computed field since this is the default value